sentry-ruby/lib/sentry/transaction.rb

```diff
@@ -70,7 +88,25 @@ def self.from_sentry_trace(sentry_trace, hub: Sentry.get_current_hub, **options)
           sampled_flag != "0"
         end
 
-      new(trace_id: trace_id, parent_span_id: parent_span_id, parent_sampled: parent_sampled, hub: hub, **options)
+      baggage = if baggage && !baggage.empty?
+                  Baggage.from_incoming_header(baggage)
+                else
+                  # If there's an incoming sentry-trace but no incoming baggage header,
+                  # for instance in traces coming from older SDKs,
+                  # baggage will be empty and frozen and won't be populated as head SDK.
+                  Baggage.new({})
+                end
+
+      baggage.freeze!
```

Comment:
Always make the baggage immutable? (Sorry if those a stupid questions, I am not as deep into the whole DS details)

Reply:
yes if sentry-trace is present, there will either be incoming baggage or no baggage (from older SDKs) and we want to freeze it in both cases. (This might change eventually when we make baggage mutable.)